### DISCUSSION TITLE
Make staticData type safe

### DISCUSSION THREAD

**Discussion by tykhan** [TanStack/router#1137] [Ideas]
Static data  of route is not type safe yet.

<img width="458" alt="image" src="https://github.com/TanStack/router/assets/62352586/820c2342-5d40-4197-ab9b-471c97b4c726">

<img width="707" alt="image" src="https://github.com/TanStack/router/assets/62352586/4cb3f4a0-0da2-4308-8b33-5a5eda280c6c">

**Comment by tannerlinsley** [comment]
It already is. https://tanstack.com/router/latest/docs/framework/react/guide/static-route-data#enforcing-static-data

**Comment by wolfgang720** [comment]
Is it possible to declare the interface for only one route or a subtree?
